pandas/tests/io/json/test_normalize.py

```diff
@@ -460,3 +487,70 @@ def test_nonetype_multiple_levels(self):
             'location.country.state.town.info.y': -33.148521423339844,
             'location.country.state.town.info.z': 27.572303771972656}
         assert result == expected
+
+    @pytest.mark.parametrize("max_level, expected", [
+        (None, [{'CreatedBy.Name': 'User001',
+                 'Lookup.TextField': 'Some text',
+                 'Lookup.UserField.Id': 'ID001',
+                 'Lookup.UserField.Name': 'Name001',
+                 'Image.a': 'b'
+                 }]),
+        (0, [{'CreatedBy': {'Name': 'User001'},
+              'Lookup': {'TextField': 'Some text',
+                         'UserField': {'Id': 'ID001', 'Name': 'Name001'}},
+              'Image': {'a': 'b'}
+              }]),
+        (1, [{'CreatedBy.Name': 'User001',
+              'Lookup.TextField': 'Some text',
+              'Lookup.UserField': {'Id': 'ID001',
+                                   'Name': 'Name001'},
+              'Image.a': 'b'
+              }])])
+    def test_with_max_level_none(self, max_level, expected):
+        # GH23843: Enhanced JSON normalize
+        data = [{
+            'CreatedBy': {'Name': 'User001'},
+            'Lookup': {'TextField': 'Some text',
+                       'UserField': {'Id': 'ID001', 'Name': 'Name001'}},
+            'Image': {'a': 'b'}
+        }]
+        output = nested_to_record(data, max_level=max_level)
+        assert output == expected
+
+    def test_with_large_max_level(self):
```

Comment:
I think rather than having this as a separate test can just parametrize the previous test (behavior should be same as None); just keeps things more succinct that way!